pkg/api/steve/supportconfigs/supportconfigs.go

```diff
@@ -0,0 +1,57 @@
+package supportconfigs
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/rancher/rancher/pkg/version"
+)
+
+// supportConfig represents the serializable struct of support related information common to all rancher setups
+type supportConfig struct {
+	// SupportEligible - if this rancher product is eligible for support.
+	SupportEligible bool `json:"support_eligible,omitempty"`
+	// Platform - the platform (arm, x86_64, etc) rancher is running on. As of now, always x86
+	Platform string `json:"platform"`
+	// Product - suse product string identifying rancher - see getRancherProductString
+	Product string `json:"product"`
+}
+
+// supportConfigHandler contains the necessary clients/controllers for retrieving a support config
+type supportConfigHandler struct {
+}
+
+// getSupportConfig retrieves a supportConfig
+func (h *supportConfigHandler) getSupportConfig() supportConfig {
+	return supportConfig{
+		SupportEligible: false,
+		Platform:        h.getRancherPlatform(),
+		Product:         h.getRancherProductString(version.Version),
+	}
+}
```

Comment:
Nit: since primary entrypoint to this method is `getSupportConfigAsMap`, I would put that as first method.